### PR TITLE
Guardian invites

### PR DESCRIPTION
**Pull request by susanxiong**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -93,6 +96,28 @@ def parse_classroom_last_used(parameters):
...
+    with suppress(ProgrammingError):
+        sql.truncate("GoogleClassroom_GuardianInvites")
+    next_page_token = ""
+    while next_page_token is not None:
+        # Get data for all students
+        results = (
+            service.userProfiles()
+            .guardianInvitations()
+            .list(
+                studentId="-", states=["PENDING", "COMPLETE"], pageToken=next_page_token
```
Comment (dchess) [inline main.py]: Do we need the states argument? Can we pull all of them?

File: main.py [hunk 2]
```diff
@@ -93,6 +96,28 @@ def parse_classroom_last_used(parameters):
...
+            )
+            .execute()
+        )
+        guardian_invites = results.get("guardianInvitations", [])
+        df = pd.json_normalize(guardian_invites)
+        df = df.astype({"creationTime": "datetime64[ns]"})
+        next_page_token = results.get("nextPageToken", None)
+        sql.insert_into("GoogleClassroom_GuardianInvites", df, if_exists="append")
```
Comment (dchess) [inline main.py]: You don't need the `if_exists="append"` argument because it's the default.

File: main.py (hunk 1 above)
Comment (susanxiong) [reply main.py]: Yes it's needed. If it's not specified, then only PENDING comes back. I'm not seeing a way to just get "all" states. But [PENDING and COMPLETE](https://developers.google.com/classroom/reference/rest/v1/userProfiles.guardianInvitations#GuardianInvitationState) are the only two.

File: main.py (hunk 2 above)
Comment (susanxiong) [reply main.py]: Ooh good to know!

**Pull request merged**